### PR TITLE
Countable pseudocharacter

### PR DESCRIPTION
**Pull request by Almanzoris**
Defined the new property "Countable pseudocharacter".

Theorems added:

- Countable pseudocharacter implies $T_1$.
- First countable $+$ $T_1$ implies countable pseudocharacter.
- $G_{\delta} + T_0$ implies countable pseudocharacter.

### CODE REVIEW

**Comment by Almanzoris** [comment]
What would be the best references to add for the definition of the property and what other theorems should be added?

**Comment by Almanzoris** [comment]
Maybe I should change the third theorem to $G_{\delta} + T_1$ and mention that it is equivalent to $G_{\delta} + T_0$?

**Comment by prabau** [comment]
Thanks for this first draft.  Here are a few comments.

In general, "$G_\delta" is short for "$G_\delta$ set" (not to be confused with "$G_\delta space").  So a sentence like "Every singleton is $G_\delta$" is not quite grammatical.  One normally says "Every singleton is a $G_\delta$" or "Every singleton is a $G_\delta$ set" (with an article in front).

Also, instead of "A space in which every singleton is a $G_\delta$, I would just say "Every point is a $G_\delta$".  It is a slight abuse of language (since intersections of sets can be the singleton containing a point, but not the point itself), but it is a very common way to expressing this, and does not cause confusion.  (Similar to the way people talk about "closed points", etc.)
(And omitting "a space in which" and giving the condition directly should be fine; we do the same for various other properties)

Exercise 3.1.F(a) in Engelking is a good reference for this.

General comment.  We need the explanations to be written in grammatical sentences.  In particular, at the beginning of T502 for example, the first sentence is "If $X$ is $G_\delta$ space and $T_0$."  That does not make much sense.  The word "if" is supposed to introduce a subordinate clause, followed by a main clause.  No main clause here.  Maybe you meant "Suppose $X$ is ...". 

Using English is clearer than using logical symbols like $\exists$ or $\mid$. What is this vertical bar supposed to mean anyway?

T502: I agree that it would be better to have [G_delta + T1] in the hypotheses.  That also matches better with T500, which says that T1 is a necessary condition for countable pseudocharacter.  No need to mention anything about T0.  pi-base already knows that G_delta space + T0 implies T1.  So any relevant example will necessarily have T1 already.

I'll add a commit to this PR for other characterizations of the T1 property that may be useful to keep in mind here.  I think some of the theorems here could be streamlined slightly.

**Comment by prabau** [comment]
Added more details for T1 property.  (turns out the previous version was not even quite correct)

@StevenClontz what name looks better?
- "Countable pseudocharacter"
- "Has countable pseudocharacter"

Did you have an alias in mind as well?

**Comment by Almanzoris** [comment]
> Thanks for this first draft. Here are a few comments.
> 
> In general, "$G_\delta" is short for "$G_\delta$ set" (not to be confused with "$G_\delta space"). So a sentence like "Every singleton is Gδ" is not quite grammatical. One normally says "Every singleton is a Gδ" or "Every singleton is a Gδ set" (with an article in front).
> 
> Also, instead of "A space in which every singleton is a Gδ, I would just say "Every point is a Gδ". It is a slight abuse of language (since intersections of sets can be the singleton containing a point, but not the point itself), but it is a very common way to expressing this, and does not cause confusion. (Similar to the way people talk about "closed points", etc.) (And omitting "a space in which" and giving the condition directly should be fine; we do the same for various other properties)
> 
> Exercise 3.1.F(a) in Engelking is a good reference for this.
> 
> General comment. We need the explanations to be written in grammatical sentences. In particular, at the beginning of T502 for example, the first sentence is "If X is Gδ space and T0." That does not make much sense. The word "if" is supposed to introduce a subordinate clause, followed by a main clause. No main clause here. Maybe you meant "Suppose X is ...".
> 
> Using English is clearer than using logical symbols like ∃ or ∣. What is this vertical bar supposed to mean anyway?
> 
> T502: I agree that it would be better to have [G_delta + T1] in the hypotheses. That also matches better with T500, which says that T1 is a necessary condition for countable pseudocharacter. No need to mention anything about T0. pi-base already knows that G_delta space + T0 implies T1. So any relevant example will necessarily have T1 already.
> 
> I'll add a commit to this PR for other characterizations of the T1 property that may be useful to keep in mind here. I think some of the theorems here could be streamlined slightly.

Alright, thank you for your clarifications. I am going to make these changes.

Yeah, I wrote it with a "," instead of "." at first, but then I separated it and I forgot to reedit... I mean it as suppose.

By "|", I meant "such that". It is true that using English is clearer, which is also an important part of the objective.

The update that you have done to the $T_1$ property has been really good and useful. I think that there are other equivalences that could be added to other separation axioms. To mention an example out of the ones that I am thinking about: Urysohn's Lemma (normal property). We could discuss this in a new issue.

I am keen on following more suggestions to simplify the theorems' proof.

**Comment by prabau** [comment]
There are two ways to preview things to double check the grammar and writing are correct before finalizing things.  (1) before a commit: within github.dev, the editor pane has a series of buttons at the top right.  One of them is for preview.  (2) after a commit is accepted by the build system, one can preview the final result from https://topology.pi-base.org: go to the Advanced pane, change the branch name from "main" to the particular branch for the PR and click Sync.

**Comment by prabau** [comment]
Regarding the T1 property, the reason I added the other characterizations is to make it easier to use in proofs of theorems.  I only wanted to list the most useful ones for that purpose. (Actually the fourth one in the list right now is not very useful and I'll remove it.)

It is true that one could add more such equivalences for other properties, for example the Urysohn lemma for normal.  But I am not sure that fits well what pi-base is really for.  Pi-base is more focused on properties and the relationships between various combinations of properties, with examples and counterexamples.  Many results in topology don't fit into this mold and don't need to be in pi-base.  That's my opinion.  But maybe others want to discuss this further.

**Comment by prabau** [comment]
One example of simplification where a expanded page for T1 makes things nearly obvious.  For T500, one could say:

"If $X$ is {P191}, every point is an intersection of open sets, which is one of the characterizations for {P2}."

**Comment by prabau** [comment]
image

**Comment by Almanzoris** [comment]
I have been shown the following:

https://math.stackexchange.com/q/240480
https://math.stackexchange.com/q/1288595

S29 is $T_1$ (and not $T_2$), $G_{\delta}$ and compact, but it isn't first countable.

**Comment by Almanzoris** [comment]
Can the comment on https://math.stackexchange.com/q/1288595 be included better in the references following the ref's format instead of citing the whole question? Or should I mention it in the description of the theorem (although that is the only comment on that question)?

**Comment by prabau** [comment]
Made a few changes.
- P191: reinstated your original formulation, which seems clearer after all
- T501: I have to say I was confused by the previous version, so changed it
- T502: some rewording to make things clearer
- T503: https://math.stackexchange.com/questions/240472/g-delta-singletons-in-compact-hausdorff-and-first-countability now also covers the locally compact case.  So only need to refer to that post.
- the new files were missing a newline on their last line (should show as an empty final line in github.dev)

Feel free to discuss anything if you have any comments.

As far as I am concerned, this is good to go, except from one last issue.  Need feedback from @StevenClontz and anybody else who wants to share an opinion.

For P91, what to choose for the main name, and some alias name?
- "Has countable pseudocharacter"
- "Has $G_\delta$ singletons"
- "Has $G_\delta$ points"
- "Countable pseudocharacter"
- something else ?

(Remark about "Has $G_\delta$ singletons/points": I think it's probably not too confusing.  Due to the plural, it's to be interpreted as all points are G_delta, and not just some point is G_delta.  (We would say "Has a $G_\delta$ point" if we meant the latter.  Similar to "Has closed retracts".)

File: properties/P000002.md [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
   - doi: 10.1007/978-1-4612-6290-9
     name: Counterexamples in Topology
 ---
 
-Given any two distinct points, each has an open set not containing the other.
+Any of the following equivalent properties holds:
+
+- Given any two distinct points, each has a neighborhood not containing the other point.
+- Points are closed in $X$ (that is, for each $x\in X$ the singleton $\{x\}$ is closed).
+- Each point is an intersection of open sets.
```
Comment (StevenClontz) [inline properties/P000002.md:18]:
```suggestion
- Each point is an intersection of its (open) neighborhoods.
```

For whatever reason, this wording feels more obvious to me.

Comment (StevenClontz) [reply properties/P000002.md:18]: Oh and it's used in T501

File: theorems/T000503.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+if:
+  and:
+  - P000130: true
+  - P000003: true
+  - P000191 : true
+then:
+  P000028: true
+refs:
+- mathse: 240480
+  name: $G_{\delta}$ singletons in compact Hausdorff and first countability
```
Comment (StevenClontz) [inline theorems/T000503.md:12]:
```suggestion
  name: Answer to "$G_{\delta}$ singletons in compact Hausdorff and first countability"
```

File: properties/P000191.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+uid: P000191
+name: Countable pseudocharacter
```
Comment (StevenClontz) [inline properties/P000191.md:3]:
```suggestion
name: Countable pseudocharacter
aliases:
  - Points-$G_\delta$
```

**Comment by StevenClontz** [comment]
> * "Countable pseudocharacter"
> * "Has countable pseudocharacter"
> 
> Did you have an alias in mind as well?

No preference on the name; I suggested an alias.

**Review by StevenClontz — COMMENTED** [review]
This is looking like a great contribution. Thanks @Almanzoris. Be sure to mark it as ready for review when you think it's ready so someone can carefully review and approve the proposed changes.

**Comment by prabau** [comment]
@StevenClontz A few comments on your latest suggestions.

For the alias name, I think "Points- $G_\delta$" is rather unclear grammatically.  " $G_\delta$ points" would make more sense.  But even better, and fine for an alias, would be "Points are $G_\delta$".

For the list of characterizations of the T1 property, please see commits 2,3,4 in the list of commits for this PR.

I added "Each point is an intersection of open sets." because it was a very simple way to express things, and it fits perfectly for the proof of T500.  No reason to remove it.

I had initially added "Every subset of $X$ is the intersection of all the open sets containing it.", but then removed it as it did not seem as helpful, and not necessary for proofs here.

Adding "Each point is the intersection of its (open) neighborhoods." as suggested should be fine (with "the intersection" instead of "an intersection").  It's used directly in the proof of T501.  Does "(open)" really need to be there?  It's kind of obvious.

**Comment by StevenClontz** [comment]
> For the alias name, I think "Points- Gδ" is rather unclear grammatically. " Gδ points" would make more sense. But even better, and fine for an alias, would be "Points are Gδ".

I believe the phrase "with points $G_\delta$" is the most common phrasing I've seen in the literature, though I'm working from memory. I'm okay with "Points are $G_\delta". I just know that I instantly recognize the meaning of those phrases, and I need to turn a few gears in my head to remember what "pseudocharacter" is.
 
> Adding "Each point is the intersection of its (open) neighborhoods." as suggested should be fine (with "the intersection" instead of "an intersection"). It's used directly in the proof of T501. Does "(open)" really need to be there? It's kind of obvious.

I'm okay removing "(open)".

**Comment by prabau** [comment]
> > For the alias name, I think "Points- Gδ" is rather unclear grammatically. " Gδ points" would make more sense. But even better, and fine for an alias, would be "Points are Gδ".
> 
> I believe the phrase "with points Gδ" is the most common phrasing I've seen in the literature, though I'm working from memory. I'm okay with "Points are $G_\delta". I just know that I instantly recognize the meaning of those phrases, and I need to turn a few gears in my head to remember what "pseudocharacter" is.

I was objecting to "Point-G_delta" as written with the hyphen, which remains grammatically unclear to me.  But no objection to "With point $G_\delta$".  It's used for example in https://zbmath.org/1349.54053.

**Comment by StevenClontz** [comment]
I've updated my suggestion accordingly.

**Comment by prabau** [comment]
Not sure if you had seen this comment earlier: https://github.com/pi-base/data/pull/695#issuecomment-2242111499, where there was a list of suggestions, and a mention specifically about the plural ...

What we want to convey is "a space where all the points are G_delta sets".  What's the best way to express this in an abbreviated form?  Please indulge the rambling below.

Wikipedia has information about [compound nouns ](https://en.wikipedia.org/wiki/English_compound) and related constructs.  Given some nouns A and B, _if "A B" and "B A" are viewed as a compound_, they are not the same thing.  "Country dog" and "dog country" are different things.  And in general in English a compound of the form "A B" means a type of B, qualified by also having the property of being an A or being related to A in some fashion.

If "G_delta point" is viewed as a compound, which seems perfectly reasonable and has the meaning of a point that is a G_delta, then "Has G_delta points" and "With G_delta points" seems ok, with the understanding that all points are G_delta.

But one could also have "With points G_delta" where "points" and "G_delta" and *not* forming a compound.  That also seems very reasonable, due the plural form "points".  Something like "As B" with "A" in the plural cannot be a compound noun.  And the only reasonable interpretation of "With As B" is "With all the As having property B".

"Has points G_delta" is maybe also ok, due to the same reason that "points" is in the plural.

What's your impression as a native speaker?

**Comment by prabau** [comment]
One benefit of "With points G_delta" over "Has points G_delta" is that it's in "adjectival form".  So the link `{P191}` would expand to something that flows better in text.

**Comment by prabau** [comment]
@StevenClontz I was probably overthinking this.  If you can confirm that "Has points G_delta" sounds the best, let's go with that as the primary name.

**Comment by StevenClontz** [comment]
My suggestion at https://github.com/pi-base/data/pull/695#discussion_r1687306933 is updated to say "Has points $G_\delta$"

**Comment by prabau** [comment]
@Almanzoris I made one last tweak to T503, assuming "weakly locally compact + T2" instead of "locally compact + T2".  It's all equivalent, but see the discussion in https://github.com/pi-base/data/pull/647#issuecomment-2118617678 and above comments for the reasoning, about not privileging P130 over P24.

@StevenClontz I also introduced a "convenience alias" of "WLC" for P23 (weakly locally compact), as it was getting tiresome to type the whole thing when doing searches in Explore.

**Comment by prabau** [comment]
@Almanzoris Maybe we did not make this very clear.  The draft state is meant to be used when the author plans to make further changes before the review.  If there are no further changes planned by the author, the PR need not go through the draft state.  Reviewers can then immediate give feedback.

Anyway, I'll make it "Ready for review" and approve now.  Thanks again.

**Review by prabau — APPROVED** [review]

**Pull request merged**